proposals/nnnn-screen-manager-play-audio.md

```diff
@@ -0,0 +1,119 @@
+
+  Only available on SDL Core 5.0+.
+
+  @param audioFile The audio file to be played by the system
+  @param tone Whether or not to play a system tone before the audio file
+*/
+- (instancetype)initWithAudioFile:(SDLFile *)audioFile playSystemTone:(BOOL)tone;
+
+/**
+  Initialize with a string to be spoken by the system speech synthesizer.
+
+  @param spokenString The string to be spoken by the system speech synthesizer
+*/
+- (instancetype)initWithSpeechSynthesizerString:(NSString *)spokenString;
+
+/**
+Initialize with a string to be spoken by the system speech synthesizer using a phonetic string.
+
+@param spokenString The string to be spoken by the system speech synthesizer
+@param phoneticType Must be one of `SAPI_PHONEMES`, `LHPLUS_PHONEMES`, `TEXT`, or `PRE_RECORDED` or no object will be created
+@param tone Whether or not to play a system tone before the synthesized speech
+*/
+- (instancetype)initWithPhoneticSpeechSynthesizerString:(NSString *)phoneticString phoneticType:(SDLSpeechCapabilities)phoneticType;
+```
+
+##### Java
+```java
+public class PlayAudioData {
+   private List<TTSChunk> prompt;
+   private SdlFile audioFile;
+
+   // All vars have getters as well but no setters
+
+   PlayAudioData(@NonNull SdlFile audioFile)
+   PlayAudioData(@NonNull String spokenString)
+   PlayAudioData(@NonNull String phoneticString, @NonNull SpeechCapabilities phoneticType)
+}
+```
+
+### Screen Manager API
+Next is the addition to the screen manager public API itself to play the audio data.
+
+##### iOS
+```objc
+@interface SDLScreenManager: NSObject
+// Everything already there
+
+/**
+Play some audio on the module.
+
+If the audio contains a file that needs to be uploaded, the file will be uploaded before it is played.
+
+The handler will be called when the module responds to the request to play.
+*/
+- (void)playAudio:(NSArray<SDLPlayAudioData *> *)audioData withCompletionHandler:(SDLScreenManagerUpdateCompletionHandler)handler;
+
+@end
+```
+
+##### Java
+```java
+public class BaseScreenManager {
+    // Everything already there
+    public void playAudio(List<PlayAudioData> audioData, CompletionListener listener)
+}
+```
+
+### JavaScript Suite APIs
+Due to the size of the iOS APIs and the similarity between the iOS, Java Suite and eventual JavaScript Suite APIs, this proposal does not present the public APIs of the JavaScript Suite APIs – especially because the JavaScript Suite APIs do not currently have a screen manager layer. The JavaScript Suite APIs should mirror the iOS and Java Suite API appropriately and is up to the maintainers' discretion. However, if any changes needed to be made such that they impacted the iOS / Java Suite API (such as the alteration, addition, or removal of a method or property), then a proposal revision would be needed.
```

Comment:
```suggestion
Due to the size of the iOS APIs and the similarity between the iOS, Java Suite and eventual JavaScript Suite APIs, this proposal does not present the public APIs of the JavaScript Suite APIs – especially because the JavaScript Suite APIs do not currently have a screen manager layer. The JavaScript Suite APIs should mirror the iOS and Java Suite API appropriately and is up to the Project Maintainer's discretion. However, if any changes needed to be made such that they impacted the iOS / Java Suite API (such as the alteration, addition, or removal of a method or property), then a proposal revision would be needed.
```